_source/_posts/2022-09-26-java-jetty-oauth.md

```diff
@@ -0,0 +1,1017 @@
+
+Start by cloning [this tutorial's GitHub repo](https://github.com/oktadev/okta-spring-boot-jetty-example):
+
+```shell
+git clone https://github.com/oktadev/okta-spring-boot-jetty-example.git
+```
+
+The repository has several projects:
+
+- `maven-jetty`: web service using Maven
+- `gradle-jetty`: web service using Gradle
+- `spring-boot-jetty-maven-no-auth`: web service using Spring Boot and Maven
+- `spring-boot-jetty-maven-okta`: web service using Spring Boot and Maven secured with Okta
+- `spring-boot-jetty-maven-auth0`: web service using Spring Boot and Maven secured with Auth0
+
+## Build a simple web service with Java and Jetty
+
+In the first part of this tutorial, you're going to create a simple web service using Maven and then Gradle. In the second half, you'll see how to upgrade this to a Spring Boot-based web service, to which you'll add JWT-based authentication and authorization using Okta and Auth0 as the OAuth 2.0 and OIDC providers.
+
+You should have the files from the GitHub repository. Open the folder `maven-jetty` in your favorite IDE.
+
+First, take a look at the `pom.xml` file (which is shown below). Notice the Jetty version is set in the properties block. The only project dependency is `jakarta.servlet-api`. It is scoped as `provided` because, in deployment, this package will be provided by the server container.
+
+The Servlet API dependency used to be `javax-servlet-api`, but Java EE is no longer maintained and has been migrated into `jakarta.servlet-api`. This happened when Java EE moved from Oracle to the Eclipse Foundation. As you'll see in the servlet code, this package provides the standardized classes and annotations used to build the servlet.
+
+The packaging type for the project is `war`, not `jar`, and the Maven War plugin has been included in the build dependencies (`maven-war-plugin`).
+
+Lastly, the Jetty Maven plugin has also been included. You can peruse [the docs here](http://www.eclipse.org/jetty/documentation/jetty-11/programming-guide/index.html#jetty-maven-plugin). It is designed to make the development and testing of Jetty servlets easy by adding Maven goals to run and test the app.
+
+```xml
+<?xml version="1.0" encoding="UTF-8"?>
+
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <groupId>com.demo</groupId>
+  <artifactId>demo</artifactId>
+  <version>1.0-SNAPSHOT</version>
+  <packaging>war</packaging>
+  <name>demo</name>
+
+  <properties>
+    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+    <maven.compiler.source>1.7</maven.compiler.source>
+    <maven.compiler.target>1.7</maven.compiler.target>
+    <jettyVersion>11.0.11</jettyVersion>
+  </properties>
+
+  <dependencies>
+    <!-- Core Servlet Package -->
+    <dependency>
+      <groupId>jakarta.servlet</groupId>
```

Comment:
Indent should be 4 spaces for XML, or 2. The most important thing is consistency in this post and the example.